free-programming-books.md

```diff
@@ -1400,6 +1401,12 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Welcome to the Java Workshop (2006)](http://javaworkshop.sourceforge.net) - Trevor Miller
 
 
+#### Codename One
+
+* [Codename One Developer Guide](https://www.codenameone.com/files/developer-guide.pdf) (PDF)
+* [Create an Uber Clone in 7 Days](http://uber.cn1.co) - Shai Almog (PDF)
```

Comment:
The discussion around "first N chapters" entries is usually "Is this a reasonably complete or useful work on its own, or is it mostly an advertisement for the full book?" and we've received PR's that suggest removal of 300 free pages partial works. I would accept this provisionally  with an added notation `(first 2 chapters)`.

Reply:
Sure. I initially wrote that but removed it because of the linter. Where would that fit in the markup?
The chapters stand on their own as a complete tutorial for building a "ToDo App" and the basics of mobile development.